<!DOCTYPE html>
<html lang="en">
  <head>
      <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta http-equiv="content-type" content="text/html; charset=utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">
    <link rel="shortcut icon" type="image/x-icon" href="/favicon.ico" />

    <title>Four years of fellowship: annual summaries for my <span class="caps">NSF</span> Graduate Research&nbsp;Fellowship</title>

    <meta charset="utf-8" />
    <meta name="citation_journal_title" content="Satoshi Village" />

    <link href="https://blog.dhimmel.com/feeds/all.atom.xml" type="application/atom+xml" rel="alternate" title="Satoshi Village Full Atom Feed" />
    <link href="https://blog.dhimmel.com/feeds/all.rss.xml" type="application/rss+xml" rel="alternate" title="Satoshi Village Full RSS Feed" />
    <link rel="stylesheet" href="https://blog.dhimmel.com/theme/css/poole.css"/>
    <link rel="stylesheet" href="https://blog.dhimmel.com/theme/css/syntax.css"/>
    <link rel="stylesheet" href="https://blog.dhimmel.com/theme/css/lanyon.css"/>
    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=PT+Serif:400,400italic,700%7CPT+Sans:400">
    <link rel="stylesheet" href="https://blog.dhimmel.com/theme/css/styles.css"/>


  <meta name="citation_date" content="2016-03-06" />



    <meta name="author" content="Daniel Himmelstein" />

    <meta name="tags" content="NSF" />
    <meta name="tags" content="National Science Foundation" />
    <meta name="tags" content="GRFP" />
    <meta name="tags" content="GRF" />
    <meta name="tags" content="fellowship" />
    <meta name="tags" content="annual activity report" />
    <meta name="tags" content="fellowship year summary" />

  </head>

  <body>
<!-- Target for toggling the sidebar `.sidebar-checkbox` is for regular
styles, `#sidebar-checkbox` for behavior. -->
<input type="checkbox" class="sidebar-checkbox" id="sidebar-checkbox">
<!-- Toggleable sidebar -->
<div class="sidebar" id="sidebar">
  <div class="sidebar-item">
    <div class="profile">
      <img src="https://blog.dhimmel.com/theme/img/profile.jpg"/>
    </div>
  </div>

  <nav class="sidebar-nav">
  <!--<a class="sidebar-nav-item" href="/"><b>Blog Home<b></a>-->
  

  <a class="sidebar-nav-item" href="/"><b>Satoshi Village Blog</b></a>
  <a class="sidebar-nav-item" href="/archives">&ndash; archives</a>
  <a class="sidebar-nav-item" href="https://dhimmel.com/"><b>Daniel Himmelstein</b></a>
  <a class="sidebar-nav-item" href="https://dhimmel.com/about">&ndash; about</a>
  <a class="sidebar-nav-item" href="https://dhimmel.com/research">&ndash; research</a>
  </nav>

  <div class="sidebar-item">
    <p>
    &copy; <a href="https://twitter.com/dhimmel">@dhimmel</a> under <a href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a>
    </p>
  </div>
</div>    <div class="wrap">
      <div class="masthead">
        <div class="container">
          <h3 class="masthead-title">
            <a href="https://blog.dhimmel.com/" title="Home">Satoshi Village</a>
            <small>the blog of Daniel Himmelstein</small>
          </h3>
        </div>
      </div>

      <div class="container content">
<div class="posts">
<div class="post">
    <h1 class="post-title">
      Four years of fellowship: annual summaries for my <span class="caps">NSF</span> Graduate Research&nbsp;Fellowship
    </h1>
    <span class="post-date">March 06, 2016</span>

    <p>As a first year graduate student at <span class="caps">UCSF</span>, I took a mandatory course titled <a href="http://coursecatalog.ucsf.edu/course/142" title="Biophysics 297 · UCSF Registrar"><em>Scientific Writing</em></a>, which helped students apply for the National Science Foundation&#8217;s <a href="https://www.nsfgrfp.org" title="NSF Graduate Research Fellowship Program Homepage">Graduate Research Fellowship</a>. I was fortunate to receive the fellowship (Grant <a href="http://grantome.com/grant/NSF/DGE-1144247" title="Grantome Record">No. 1144247</a>), which has funded the bulk of my PhD since its third&nbsp;year.</p>
<p>At the end of each fellowship year, fellows submit an Annual Activities Report, which includes a written Fellowship Year Summary. Below I&#8217;ve reproduced my summaries from the last four years. But first, here&#8217;s a screenshot of the&nbsp;prompt:</p>
<p><img alt="Screenshot of the Fellowship Year Summary prompt" src="https://blog.dhimmel.com/nsf-year-summaries/prompt.png" title="Prompt for the Fellowship Year Summary"></p>
<h2 id="summary-of-201213-dated-april-1-2013">Summary of 2012–13 dated April 1,&nbsp;2013</h2>
<p>The 2012–2013 reporting period spans my first and second years of graduate school. The first year of graduate school consisted of coursework, seminars, and rotations. My first rotation, supervised by Dr. Ryan Hernandez, focused on human evolution and population genomics. I simulated sequencing data capturing the complexity of human demographic history. During my second rotation with Dr. John Witte, I compared the prevalence of gene-gene associations with disease under additive and epistatic disease models. The project, related to my original <span class="caps">NSF</span> proposal, resulted in a poster presentation at the <em>American Society of Human Genetics Annual Meeting</em>. In my third rotation, with Dr. Sergio Baranzini, I worked on modelling biological entities using heterogeneous networks. The project consisted of parsing and integrating publicly available databases. Professor Baranzini is now my advisor. My project aims to discover new uses for existing drugs using a network-based computational link prediction&nbsp;technique.</p>
<p>As I mentioned in my Personal Statement, I think science could be improved by better communication. Communication includes a broad range of issues ranging from writing style to open-access publishing. Over the past year, I’ve stressed my own ability to communicate. Professor Hernandez requested me as the teaching assistant for a new course he was teaching called <em>Computational Evolutionary Genomics</em>. As part of the quarter long course, I led weekly discussion sections. I’ve also honed my communication skills through presentations to lab members, students, and&nbsp;faculty.</p>
<p>Over the last year, I’ve begun to apply my interest in high-throughput computationally intensive problems to medicine. Through my research on drug repurposing, I not only hope to find new disease therapies, but also spread my scientific methodologies through improved scientific&nbsp;communication.</p>
<h2 id="summary-of-201314-dated-march-31-2014">Summary of 2013–14 dated March 31,&nbsp;2014</h2>
<p><strong>Intellectual Merit</strong>: Over the past year, my research has matured extensively. My project aims to identify genes associated with complex human diseases. My findings will help with disease susceptibility prediction and understanding what biological characteristics underlie pathogenic variants. The cross-disciplinary method improves on a technique first developed for social network analysis and relies on advanced machine learning and distributed computing. I applied for and received a $2,000 <em><span class="caps">UCSF</span> Graduate Student Research Fellowship</em> to add <span class="caps">192GB</span> of low-latency memory to our <span class="caps">HPC</span> workstation. In the past year, I exhibited the work at the <em>American Society of Human Genetics Annual Meeting</em> and presented it at the <em>Biomedical Computation at Stanford Symposium</em>. I also discuss the project in a <a href="https://youtu.be/GPAvruPpZ3A" title="YouTube · ISMB 2014 Late Breaking Research Proposal">public youtube video</a> that I&nbsp;created.</p>
<p><strong>Broader Impacts</strong>: My vision for science is one of greater accessibility and communication. I helped other students achieve proficiency in written communication as a teaching assistant for our program&#8217;s <em>Writing Predoctoral Fellowships</em> course. Personally, I assisted with peer-reviewing a journal manuscript and writing a grant proposal. In October, I traveled to Guatemala to participate in a <em>Forum on Innovation in Medical Education</em>. Hosted by a private, forward-thinking University, <span class="caps">UFM</span>, the forum focused on designing a new medical degree program based on embracing disruptions. Throughout the forum I suggested ways for strong communication and accessibility to be embedded into the structure of the new curriculum. I am an outspoken advocate of open access publishing allowing unlimited reuse with attribution. I also support developments to make science more transparent and incremental. To that end, I have developed a <a href="http://dhimmel.com/" title="Homepage of Daniel Himmelstein">personal website</a> describing my research and have begun posting material to <a href="https://figshare.com/authors/Daniel_Himmelstein/523637" title="Daniel Himmelstein on figshare">figshare</a>.</p>
<h2 id="summary-of-201415-dated-march-18-2015">Summary of 2014–15 dated March 18,&nbsp;2015</h2>
<p><strong>Intellectual Merit</strong>: My research focuses on making novel biological predictions from heterogeneous networks. Heterogeneous networks contain multiple entity and relationship types making them an ideal approach for data integration. Applying our edge prediction algorithm to a multiscale network of pathogenesis, we successfully predicted disease-associated genes, uncovered multiple sclerosis risk genes, and highlighted the mechanisms underlying pathogenesis. This study is <a href="https://doi.org/10.1101/011569" title="bioRxiv · Heterogeneous Network Edge Prediction: A Data Integration Approach to Prioritize Disease-Associated Genes">currently available</a> as a preprint. In July, I gave my first talk at a major international conference, when I presented this research at the <em>Intelligent Systems in Molecular Biology</em> conference in Boston, supported by an <span class="caps">NSF</span>-funded travel fellowship. Furthermore, our study was <a href="http://biomedicalcomputationreview.org/content/unlocking-genetics-complex-diseases-gwas-and-beyond" title="Unlocking the Genetics of Complex Diseases: GWAS and Beyond">featured</a> in the <em>Biomedical Computation Review</em>, a quarterly magazine produced by&nbsp;Stanford.</p>
<p>Next, we plan to identify new uses for existing drugs using our network-based approach. We hope to bring new life to the sluggish and increasingly expensive drug discovery process. Interested parties can <a href="https://doi.org/10.15363/thinklab.4" title="Repurposing drugs on a heterogeneous network">follow and contribute</a> to this project in&nbsp;real-time.</p>
<p><strong>Broader Impacts</strong>: In the last year, I have taken many steps to move science closer to my vision of openness. Realizing that science moves faster and reproducibility improves when source code is open, I made 156 public <a href="https://github.com/dhimmel" title="Daniel Himmelstein on GitHub">contributions on GitHub</a>. Additionally, I created a website (<a href="http://het.io/" title="Heterogeneous networks in bioinformatics">het.io</a>) covering our research on heterogeneous networks. We provide our predictions of disease-associated genes through an interactive <a href="http://het.io/disease-genes/browse/" title="hetio · Disease-Associated Gene Browser">browser</a>. I also started a <a href="http://blog.dhimmel.com/" title="Satoshi Village · the blog of Daniel Himmelstein">personal blog</a> that aims to communicate science to a broad readership. Similarly, <span class="caps">UCSF</span> hosted a <em>three minute thesis</em> competition, where students present their dissertation research with a general audience in mind. Although my submission was not selected as a finalist, I posted the <a href="https://youtu.be/oV3Wx14E7WE" title="YouTube · 2014 Three Minute Thesis Entry at UCSF">recording online</a>. <span class="caps">UCSF</span> also hosts a seminar series that accepts student speaker nominations. My nomination of Lior Pachter — professor at <span class="caps">UC</span> Berkeley and author of the <a href="https://liorpachter.wordpress.com/" title="Bits of DNA"><em>Bits of <span class="caps">DNA</span></em></a> blog — was accepted and I was fortunate to arrange his visit and discuss science with him. Finally, I performed my first <a href="https://publons.com/review/65947/" title="Publons review of 'Optimal Algorithm for Metabolomics Classification and Feature Selection varies by Dataset'">solo peer review</a> for the open access journal <em><span class="caps">PLOS</span> One</em>.</p>
<p>At a <a href="https://peerj.com/blog/post/97879213303/save-the-date-participative-bay-area-oa-week-event-for-generation-open/" title="PeerJ blog · Bay Area OA week event for Generation Open">Bay Area open access event</a>, I met the founder of <a href="https://think-lab.github.io/" title="Thinklab · a community for real-time open collaborative science">Thinklab</a>, a nascent platform for massively collaborative open science. Our project on drug repurposing is one of the initial projects using the platform. Thinklab has funded our proposal and rewards community participants for contributing to the project. We have high hopes that the <a href="https://think-lab.github.io/blog/introducing-thinklab-a-platform-for-massively-collaborative-open-science/38/" title="Introducing ThinkLab — A platform for massively collaborative open science">Thinklab model</a> can accelerate science, help reproducibility, and encourage transparent&nbsp;research.</p>
<h2 id="summary-of-201516-dated-march-6-2016">Summary of 2015–16 dated March 6,&nbsp;2016</h2>
<p><strong>Intellectual Merit</strong>: For the last year, I&#8217;ve been constructing a network for drug repurposing. We will use the network to predict the probability that a given drug–disease pair is an efficacious treatment. Since the network contains multiple node and relationships types, it&#8217;s called a hetnet — a term we&#8217;re <a href="https://doi.org/10.15363/thinklab.d104" title="Thinklab discussion · Renaming ‘heterogeneous networks’ to a more concise and catchy term">formally popularizing</a>. Our hope is that by integrating diverse data types, the hetnet will form a foundation for powerful&nbsp;analyses.</p>
<p>We&#8217;ve <a href="https://doi.org/10.15363/thinklab.d112" title="Thinklab discussion · Using the neo4j graph database for hetnets">migrated</a> to using the Neo4j graph database to store and operate on our hetnets. Leveraging this mature open source technology has improved our interoperability and provided new capabilities. Our <a href="http://portal.graphgist.org/graph_gists/drug-repurposing-by-hetnet-relationship-prediction-a-new-hope" title="Neo4j GraphGist · Drug repurposing by hetnet relationship prediction: a new hope">interactive tutorial</a> showcases our project and use of the Neo4j&nbsp;technology.</p>
<p>To build our hetnet, we created several intermediate resources. We&#8217;re particularly excited about three resources that we think the community will appreciate. First, we created a catalog of medical indications, which can be used to train and evaluate computational methods. The catalog was created by integrating four existing resources. However, we <a href="https://doi.org/10.15363/thinklab.d95" title="Thinklab discussion · Expert curation of our indication catalog for disease-modifying treatments">recruited</a> three physicians to review each drug–disease pair and classify it as disease-modifying or symptomatic. Second, we computed <a href="https://doi.org/10.15363/thinklab.d43#6" title="Thinklab discussion · Computing consensus transcriptional profiles for LINCS L1000 perturbations">consensus signatures</a> for <span class="caps">LINCS</span> L1000, which profiles transcriptional responses to perturbation. Third, we created a <a href="https://git.dhimmel.com/gene-ontology/" title="User-friendly and customizable Gene Ontology annotations">user-friendly webapp</a> for retrieving Gene Ontology&nbsp;annotations.</p>
<p><strong>Broader Impacts</strong>: To construct our hetnet, we integrated 28 public resources. Even though the resources were publicly funded, copyright and access agreements created a major licensing headache. After a <a href="https://doi.org/10.15363/thinklab.d107" title="Thinklab discussion · Integrating resources with disparate licensing into an open network">7,500+ word discussion</a> involving lawyers and other experts, we settled on the least bad way&nbsp;forward.</p>
<p>Since undergoing this copyꞃightmare and realizing that a crisis looms, I&#8217;ve become an advocate for data copyright and licensing reform in science. I presented on the legal problems related to data reuse at <a href="https://youtu.be/2Y2OIutYE4Y" title="YouTube · Copyright vs open science - Project Presentation at OpenCon"><em>OpenCon</em> in Belgium</a> and at a local <a href="https://youtu.be/Jy-E7PheIRs" title="YouTube · Realtime Open Science on Thinklab &amp; The Horrors of Data Copyright">Open Access Week event</a>.</p>
<p>I&#8217;ve been performing realtime, open notebook science using Thinklab and GitHub. Our drug repurposing <a href="http://doi.org/10.15363/thinklab.4" title="Thinklab Project · Repurposing drugs on a hetnet">project on Thinklab</a> has generated 60 discussions containing 346 comments by 8 project team members and 29 community reviewers. Getting feedback in the early stages has strengthened our project while helping us avoid poor uses of&nbsp;time.</p>
<p>In the last year, I reviewed for <a href="https://publons.com/review/245916/" title="Publons review"><em><span class="caps">PLOS</span> Computational Biology</em></a> and the <a href="https://publons.com/review/173426/" title="Publons review"><em>Pacific Symposium on Biocomputing</em></a>; authored 906 public <a href="https://github.com/dhimmel" title="Daniel Himmelstein on GitHub">GitHub contributions</a>; mentored an apprentice and an intern; and crafted a <a href="https://slides.com/dhimmel/data-biologist-cookbook" title="Slides · The Data Biologist Cookbook">data biologist cookbook</a> for the new&nbsp;students.</p>
<p>Finally, while waiting for my accepted <a href="https://doi.org/10.1371/journal.pcbi.1004259" title="Heterogeneous Network Edge Prediction: A Data Integration Approach to Prioritize Disease-Associated Genes">paper</a> to be published, I grew frustrated by the delays. I compared the delays at several journals in my field and <a href="http://blog.dhimmel.com/plos-and-publishing-delays/" title="Publication delays at PLOS and 3,475 other journals">blogged</a> the findings. <a href="https://doi.org/10.1038/523131f" title="Long wait for publication plagues many journals"><em>Nature News</em></a> and <a href="http://thepublicationplan.com/2015/07/02/publication-and-acceptance-times-for-over-3000-journals/" title="Publication and acceptance times for over 3,000 journals"><em>The Publication Plan</em></a> covered my findings. Later I performed a more <a href="http://blog.dhimmel.com/history-of-delays/" title="The history of publishing delays">comprehensive analysis</a> on the history of publishing delays, which was also covered by <a href="https://doi.org/10.1038/530148a" title="The Waiting Game · Does it take too long to publish research?"><em>Nature News</em></a> and <a href="http://thepublicationplan.com/2016/02/19/are-publishing-delays-getting-worse/" title="Are publishing delays getting worse?"><em>The Publication Plan</em></a>. My goal with this research is to help researchers avoid excessive delays while replacing anecdote with evidence in the contemporary discussion of scientific&nbsp;publishing.</p>
<h2 id="reflections">Reflections</h2>
<p>I&#8217;ll close with four&nbsp;reflections:</p>
<ol>
<li>My productivity exceeded linear growth over the four year period. Experience has played a role, in addition to the rise of open source tools for data&nbsp;science.</li>
<li>The number of hyperlinks per year has grown rapidly from 0 to 3 to 13 to 21. I attribute much of this increase to open science, which has led to more public output per unit of research&nbsp;time.</li>
<li>The deadline has always been May 1, yet each year I&#8217;ve finished the summary further in advance than the year before. Am I becoming less&nbsp;just-in-time?</li>
<li>I have yet to &#8220;discover new uses for existing drugs&#8221; despite aiming for this in 2013. Let&#8217;s make 2016 the&nbsp;year!</li>
</ol>

    <h2 id="comments">Comments</h2>
<div id="disqus_thread"></div>
<script type="text/javascript">
    var disqus_shortname = 'satoshivillage';
    var disqus_title = 'Four years of fellowship: annual summaries for my <span class="caps">NSF</span> Graduate Research&nbsp;Fellowship';

    (function() {
        var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
        dsq.src = 'https://' + disqus_shortname + '.disqus.com/embed.js';
        (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>

  </div>
</div>
      </div>

      <label for="sidebar-checkbox" class="sidebar-toggle"></label>

      <script>
        /* Lanyon & Poole are Copyright (c) 2014 Mark Otto. Adapted to Pelican 20141223 and extended a bit by @thomaswilley */
        (function(document) {
          var toggle = document.querySelector('.sidebar-toggle');
          var sidebar = document.querySelector('#sidebar');
          var checkbox = document.querySelector('#sidebar-checkbox');
          document.addEventListener('click', function(e) {
            var target = e.target;
            if(!checkbox.checked ||
            sidebar.contains(target) ||
            (target === checkbox || target === toggle)) return;
            checkbox.checked = false;
            }, false);
            })(document);
      </script>

<script type="text/javascript">
    var pkBaseURL = (("https:" == document.location.protocol) ? "https://piwik.dhimmel.com/" : "http://piwik.dhimmel.com/");
document.write(unescape("%3Cscript src='" + pkBaseURL + "piwik.js' type='text/javascript'%3E%3C/script%3E"));
</script><script type="text/javascript">
try {
var piwikTracker = Piwik.getTracker(pkBaseURL + "piwik.php", 5);
piwikTracker.trackPageView();
piwikTracker.enableLinkTracking();
} catch( err ) {}
</script><noscript><p><img src="http://piwik.dhimmel.com/piwik.php?idsite=5" style="border:0" alt="" /></p></noscript>


     </div>
  </body>
</html>